libs/lakebase/connect.go

```diff
@@ -5,16 +5,90 @@ import (
 	"errors"
 	"fmt"
 	"os"
+	"os/exec"
 	"strings"
+	"time"
 
 	"github.com/databricks/cli/libs/cmdctx"
 	"github.com/databricks/cli/libs/cmdio"
-	"github.com/databricks/cli/libs/exec"
+	execlib "github.com/databricks/cli/libs/exec"
 	"github.com/databricks/databricks-sdk-go/service/database"
 	"github.com/google/uuid"
 )
 
+const (
+	// Default retry configuration
+	defaultMaxRetries    = 3
+	defaultInitialDelay  = 1 * time.Second
+	defaultMaxDelay      = 10 * time.Second
+	defaultBackoffFactor = 2.0
+)
+
+// RetryConfig holds configuration for connection retry behavior
+type RetryConfig struct {
+	MaxRetries    int
+	InitialDelay  time.Duration
+	MaxDelay      time.Duration
+	BackoffFactor float64
+	Enabled       bool
+}
+
+// isRetryableConnectionError checks if the error output indicates a retryable connection issue
+func isRetryableConnectionError(output []byte) bool {
```

Comment:
go SDK has IsRetriable(), can you add a comment why that cannot or should not be used for this case?

https://github.com/databricks/databricks-sdk-go/blob/4c75570d922a076d4e0f98ced457728de3d23188/apierr/errors.go#L95

There is also retries package, same question: https://github.com/databricks/databricks-sdk-go/tree/4c75570d922a076d4e0f98ced457728de3d23188/retries